### PR TITLE
use class for ancestors and remove set parent as current

### PR DESCRIPTION
**Pull request by andrey1s**
https://github.com/KnpLabs/KnpMenuBundle/issues/331#issuecomment-327149795

## Changelog
```markdown
### Changed
 - add specific class for ancestors of a current item `ancestorClass: "active"`

### Removed
 - `Sonata\AdminBundle\Menu\Matcher\Voter\CurrentVoter` using a class only for the current elements
```
## Subject

- add render specific class "active" for ancestors 
- remove KnpMenu CurrentVoter
- remove test to CurrentVoter

### CODE REVIEW

**Comment by OskarStark** [comment]
Please associate your commit with your github email address, thank you!

**Comment by greg0ire** [comment]
Wow this is so much more concise

File: Menu/Matcher/Voter/ChildrenVoter.php [hunk 1]
```diff
@@ -1,61 +0,0 @@
-<?php
```
Comment (core23) [inline Menu/Matcher/Voter/ChildrenVoter.php]: This is a BC break

**Comment by stof** [comment]
This will have another benefit: it will fix https://github.com/sonata-project/SonataAdminBundle/issues/4805 as it will remove the voter needing to get the matcher inside it (while the matcher itself needs the voter)

**Comment by greg0ire** [comment]
Hmm, no reactions, I think I'm going to carry it.

File: Menu/Matcher/Voter/ChildrenVoter.php (hunk 1 above)
Comment (stof) [reply Menu/Matcher/Voter/ChildrenVoter.php]: this special voter is breaking the concept of `current` item for people using KnpMenu in their project outside the admin part. So removing the voter actually fixes bugs.

Comment (greg0ire) [reply Menu/Matcher/Voter/ChildrenVoter.php]: I think I'll keep the class around but remove the service, and deprecate the class.

**Comment by greg0ire** [comment]
See #4806

**Pull request closed without merging**